test/e2e/cluster_upgrade.go

```diff
@@ -157,6 +157,7 @@ func ClusterUpgradeSpec(ctx context.Context, inputGetter func() ClusterUpgradeSp
 			ClusterProxy:                input.BootstrapClusterProxy,
 			Cluster:                     result.Cluster,
 			ControlPlane:                result.ControlPlane,
+			UpgradeMachineTemplate:      ptr.To(fmt.Sprintf("%s-control-plane-1.30", clusterName)),
```

Comment:
Does it feel odd to hard-code these version numbers into the test?  I suppose it's fine b/c it could be any version right?  like 1.xx

Reply:
Yes, we don't test a particular version but rather if the upgrade mechanism of the provider is working as expected.